Implement debugging of iOS Hermes application (#1488)
microsoft/vscode-react-native: README.md

The extension allows you to debug multiple devices and configurations, please read the following sections for more information for your particular use case.

## Hermes engine

> Hermes is the new JavaScript engine optimized for running React Native apps on Android. It improves app performance and decreases app size.

Reviewer: ```suggestion
Hermes is the new JavaScript engine optimized for running React Native apps on Android and iOS. It improves app performance and decreases app size.
```

}
```

### iOS Hermes

> The extension allows you to debug iOS Hermes applications. See [iOS Hermes debugging](#ios-hermes-debugging) for more details.

Reviewer: ```suggestion
The extension provides experimental support of debugging iOS Hermes applications. See [iOS Hermes debugging](#ios-hermes-debugging) for more details.
```

### Attach to Hermes application

> To attach to the running Hermes application use `Attach to Hermes application - Experimental` launch configuration:

Reviewer: ```suggestion
To attach to a running Hermes application use `Attach to Hermes application - Experimental` launch configuration:
```

### iOS direct debugging

> The extension provides experimental support of iOS direct debugging. At the moment direct iOS debugging only works with applications running on the device. See more info here: [react-native-community/discussions-and-proposals#40](https://github.com/react-native-community/discussions-and-proposals/issues/40), [react-native-community/discussions-and-proposals#206](https://github.com/react-native-community/discussions-and-proposals/issues/206)

Reviewer: ```suggestion
The extension provides experimental support of iOS direct debugging. See more info here: [react-native-community/discussions-and-proposals#40](https://github.com/react-native-community/discussions-and-proposals/issues/40), [react-native-community/discussions-and-proposals#206](https://github.com/react-native-community/discussions-and-proposals/issues/206)

For now the extension supports iOS direct debugging only on real iOS devices.
```

    "port": 9221
```

### iOS Hermes debugging

> You can enable Hermes engine for iOS application by edit `ios/Podfile` file and make the change illustrated below:

Reviewer: ```suggestion
You can enable Hermes engine for an iOS application by editing `ios/Podfile` file the following way:
```

> ```json

Reviewer: ```suggestion
```diff
```

-  use_react_native!(:path => config[:reactNativePath])
+  use_react_native!(:path => config[:reactNativePath], :hermes_enabled => true)
```

> After this changes you need to execute `pod install` in `ios` folder. And now you can use `Debug iOS Hermes - Experimental` launch configuration to debug iOS Hermes application:

Reviewer: ```suggestion
After this change you need to execute `pod install` command in `ios` folder. After that you can use `Debug iOS Hermes - Experimental` launch configuration to debug an iOS Hermes application:
```
